internal/controller/serviceaccount_controller.go

```diff
@@ -471,3 +475,15 @@ func (r *serviceAccountReconciler) detachImagePullSecret(
 
 	return nil
 }
+
+func resolvePrincipals(sa *corev1.ServiceAccount) []string {
+	// Note: AWS and Google providers cannot be used together since they share
+	// the imagepullsecrets.preferred.jp/registry annotation. This function
+	// returns the first non-empty annotation found.
+	for _, key := range []string{annotationKeyGoogleSA, annotationKeyAWSRoleARN} {
```

Comment:
In the current implementation AWS has precedence over Google. This line should align with the order too.

https://github.com/pfnet/image-pull-secrets-provisioner/blob/694b9d187b582a23240467911fe81d26d0b5dca4/internal/controller/serviceaccount_controller.go#L343-L359